flexget/plugins/input/from_telegram.py

```diff
@@ -0,0 +1,219 @@
+                    'url': {'type': 'string', 'format': 'regex'},
+                    'title': {'type': 'string', 'format': 'regex'},
+                },
+                'additionalProperties': {'type': 'string', 'format': 'regex'},
+            },
+        },
+        'required': ['token'],
+        'additonalProperties': False,
+    }
+
+    @plugin.internet(logger)
+    def on_task_input(self, task, config):
+        #Load The Configs
+        token = config['token']
+        only_new = config['only_new']
+        entry_config = config.get('entry')
+        whitelist = config.get('whitelist',[])
+        types = config.get('types')
+
+        #Defaults
+        if not types:
+            types = ['private', 'group']
+
+        #Get Last Checked ID
+        update_id = task.simple_persistence.get(token+'_update_id')
+
+        #Get only new messages
+        params={}
+        if update_id and only_new:
+            update_id += 1
+            params['offset'] = update_id
+        
+        #The Target URL
+        url = "{domain}bot{token}/getUpdates".format(
+            domain=_TELEGRAM_API_URL,
+            token=token
+        )
+
+        #Get Telegram Updates
+        try:
+            response = task.requests.get(
+                url, timeout=60, raise_status=False, params=params
+            ).json()
+        except HTTPError as e:
+            raise plugin.PluginError("Error getting telegram update: {}" % e)
+            return
+
+        #We have a error
+        if not response['ok']:
+            raise plugin.PluginError("Telegram updater returned error {}: {}" % (response['error_code'], response['description'] ) )
+            return
+
+        #Get All New Messages
+        messages = response['result']
+
+        entries = []
+        for message in messages:
+            #This is the ID
+            update_id = message['update_id']
+
+            #Update the last ID for the Bot
+            logger.debug("Last Update set to {}",update_id)
+            task.simple_persistence[token+'_update_id'] = update_id
+            
+            #We Don't care if it's not a message or no text
+            if 'message' not in message or 'text' not in message['message'] or 'chat' not in message['message'] or 'type' not in message['message']['chat']:
+                logger.debug("Invalid message discarted: {}", message)
+                continue
+
+            logger.debug("Income message: {}", message)
+
+            #Create Base Entry
+            text = message['message']['text']
+            entry = Entry()
+            entry['title'] =  text
+
+            #We need a url, so we add a dummy
+            entry['url'] = "http://localhost?update_id="+str(update_id)
+
+            #Store the message if we need to use it in other plugins
+            entry['telegram_message'] =  message['message']
+            
+            #Check Types
+            if types and message['message']['chat']['type'] not in types:
```

Comment:
this check should probably happen sooner

Reply:
A bit sooner! Done!